core/templates/dev/head/domain/story_viewer/story-viewer-backend-api.service.ts

```diff
@@ -16,59 +16,78 @@
  * @fileoverview Service to get story data.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-require('domain/story_viewer/story-viewer-domain.constants.ajs.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('StoryViewerBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService', 'STORY_DATA_URL_TEMPLATE',
-  'STORY_PROGRESS_URL_TEMPLATE', function(
-      $http, $q, UrlInterpolationService, STORY_DATA_URL_TEMPLATE,
-      STORY_PROGRESS_URL_TEMPLATE) {
-    var storyDataDict = null;
-    var _fetchStoryData = function(storyId, successCallback, errorCallback) {
-      var storyDataUrl = UrlInterpolationService.interpolateUrl(
-        STORY_DATA_URL_TEMPLATE, {
-          story_id: storyId
-        });
+import { StoryViewerDomainConstants } from
+  'domain/story_viewer/story-viewer-domain.constants';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
 
-      $http.get(storyDataUrl).then(function(response) {
-        storyDataDict = angular.copy(response.data);
-        if (successCallback) {
-          successCallback(storyDataDict);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
-    };
+@Injectable({
+  providedIn: 'root'
+})
+export class StoryViewerBackendApiService {
+  // TODO(#7176): Replace 'any' with the exact type. This has been kept as
+  // 'any' because 'storyDataDict' is a dict with underscore_cased
+  // keys which give tslint errors against underscore_casing in favor of
+  // camelCasing.
+  storyDataDict: any = null;
+
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private http: HttpClient
+  ) {}
 
-    var _recordChapterCompletion = function(
-        storyId, nodeId, successCallback, errorCallback) {
-      var chapterCompletionUrl = UrlInterpolationService.interpolateUrl(
-        STORY_PROGRESS_URL_TEMPLATE, {
-          story_id: storyId,
-          node_id: nodeId
-        });
-      $http.post(chapterCompletionUrl).then(function(response) {
-        successCallback({
-          summaries: response.data.summaries,
-          nextNodeId: response.data.next_node_id,
-          readyForReviewTest: response.data.ready_for_review_test});
+  // Return type is any because this function doesn't actually return anything
+  _fetchStoryData(storyId: string, successCallback, errorCallback): any {
```

Comment:
If it doesn't return anything, you can add the returnType as ```void```. Also I believe the type of successCallback and errorCallback is ```Function```

Reply:
Done!